### PR TITLE
[Refactor] Convert tests and replace ZMGenericMessage

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?
To replace ZMGenericMessage with GenericMessage, we need to convert the tests to Swift.
Tests classes that were converted:
 ConversationTests, FileTransferTests, SendAndReceiveMessagesTests, MessagingTests

### CODE REVIEW

File: Tests/Source/Integration/ConversationTests+Archiving.swift [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        // when
+        self.mockTransportSession.performRemoteChanges { (session) in
+            block(session)
+        }
+        XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+
+        // then
+        shouldUnarchive
+            ? XCTAssertFalse(conversation!.isArchived)
+            :  XCTAssertTrue(conversation!.isArchived)
```
Comment (typfel) [inline Tests/Source/Integration/ConversationTests+Archiving.swift:52]:
```suggestion
            ? XCTAssertFalse(conversation!.isArchived)
            : XCTAssertTrue(conversation!.isArchived)
```

File: Tests/Source/Integration/DeleteMessagesTests.swift [hunk 2]
```diff
@@ -106,9 +106,9 @@ class DeleteMessagesTests: ConversationTestsBase {
         let textMessage = GenericMessage(content: Text(content: "Hello"))
         
         // when
-        mockTransportSession.performRemoteChanges { session in
+        self.mockTransportSession.performRemoteChanges { session in
             do {
-                self.selfToUser1Conversation.encryptAndInsertData(from: firstClient, to: selfClient, data: try textMessage.serializedData())
+                self.groupConversation.encryptAndInsertData(from: firstClient, to: selfClient, data: try textMessage.serializedData())
```
Comment (typfel) [inline Tests/Source/Integration/DeleteMessagesTests.swift:111]: Shouldn't this still be `selfToUser1Conversation`?

File: Tests/Source/Integration/DeleteMessagesTests.swift [hunk 3]
```diff
@@ -123,10 +123,9 @@ class DeleteMessagesTests: ConversationTestsBase {
         
         let genericMessage = GenericMessage(content: MessageDelete(messageId: message.nonce!))
         
-        // when
-        mockTransportSession.performRemoteChanges { session in
+        self.mockTransportSession.performRemoteChanges { session in
             do {
-                self.selfToUser1Conversation.encryptAndInsertData(from: secondClient, to: selfClient, data: try genericMessage.serializedData())
+                self.groupConversation.encryptAndInsertData(from: secondClient, to: selfClient, data: try genericMessage.serializedData())
```
Comment (typfel) [inline Tests/Source/Integration/DeleteMessagesTests.swift:128]: Shouldn't this still be `selfToUser1Conversation`?

File: Tests/Source/Integration/SlowSyncTests+Swift.swift [hunk 4]
```diff
@@ -66,7 +67,7 @@ class SlowSyncTests_Swift: IntegrationTest {
     
     func testThatItDoesAQuickSyncAfterTheWebSocketWentDown() {
         // GIVEN
-        XCTAssertTrue(login())
+        XCTAssertTrue(self.login())
```
Comment (typfel) [inline Tests/Source/Integration/SlowSyncTests+Swift.swift:70]: According to our coding conventions we only add `self` when necessary.

File: Tests/Source/Integration/SlowSyncTests+Swift.swift [hunk 5]
```diff
@@ -28,7 +29,7 @@ public extension AssetRequestFactory {
     }
 }
 
-class SlowSyncTests_Swift: IntegrationTest {
+class SlowSyncTests_Swift: ConversationTestsBase {
```
Comment (typfel) [inline Tests/Source/Integration/SlowSyncTests+Swift.swift:32]: How come this needs to be subclass of `ConversationTestsBase` now?

Comment (typfel) [reply Tests/Source/Integration/SlowSyncTests+Swift.swift:32]: Realised that `ConversationTestsBase` is a new class when reviewing David's PR. So guess we depend on a helper method from that base class.

Comment (KaterinaWire) [reply Tests/Source/Integration/SlowSyncTests+Swift.swift:32]: `ConversationTestsBase` has a setUp that `IntegrationTest` doesn't have

File: Tests/Source/Integration/SlowSyncTests+Swift.swift (hunk 4 above)
Comment (KaterinaWire) [reply Tests/Source/Integration/SlowSyncTests+Swift.swift]: Fixed

File: Tests/Source/Integration/DeleteMessagesTests.swift (hunk 2 above)
Comment (KaterinaWire) [reply Tests/Source/Integration/DeleteMessagesTests.swift:111]: It was originally as self.groupConversation.

File: Tests/Source/Integration/DeleteMessagesTests.swift (hunk 3 above)
Comment (KaterinaWire) [reply Tests/Source/Integration/DeleteMessagesTests.swift:128]: It was originally as self.groupConversation.

**Pull request merged**